ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -1247,15 +1152,17 @@ def obstacle_cb(self, msg):
 
     def send_waypoints(self):
         # generates the list of LOOKAHEAD_WPS waypoints based on car location
-        # for now assume waypoints form a loop - may not be the case
 
-        # do this at start of each cycle so that it doesn't change 
+        # do this at start of each cycle so that it doesn't change
         # if traffic cb happens in middle of loop
         self.next_tl_wp = self.next_tl_wp_tmp
```

Comment:
Now that the car has the possibility of continuing to drive through a tl_wp if stopping distance is too short, we may want to add a validation check on assigning `self.next_tl_wp` to make sure `self.next_tl_wp_tmp >= self.final_waypoints_start_ptr` and if not just force it to = -1, so this would also have to move down below line 1161.  TL detector node should update the waypoint back to -1 after driving past it but there might be some delay due to light state validation logic or other reason.

Reply:
Done